File: 2015/day3/go/mover/mover_test.go
package mover

import (
	"reflect"
	"testing"
)

func TestNewLocation(t *testing.T) {
	cases := []struct {
		in struct {
			location  []int
			direction rune
		}
		want []int
	}{
		{struct {
			location  []int
			direction rune
		}{[]int{0, 0}, '^'}, []int{0, 1}},
		{struct {
			location  []int
			direction rune
		}{[]int{0, 0}, 'v'}, []int{0, -1}},
		{struct {
			location  []int
			direction rune
		}{[]int{0, 0}, '>'}, []int{1, 0}},
		{struct {
			location  []int
			direction rune
		}{[]int{0, 0}, '<'}, []int{-1, 0}},
	}

	for _, c := range cases {
		got := NewLocation(c.in.location, c.in.direction)
		if reflect.DeepEqual(got, c.want) == false {
			t.Errorf("NewLocation(%v) == %v, want %v", c.in, got, c.want)
		}
	}
}

func TestTakeDirections(t *testing.T) {
	cases := []struct {
		in   string
		want int
	}{
		{">", 2},
		{"^>v<", 4},
		{"^v^v^v^v^v", 2},
	}

	for _, c := range cases {
		got := TakeDirections(c.in)
		if got != c.want {
			t.Errorf("TakeDirections(%v) == %d, want %d", c.in, got, c.want)
		}
	}
}

File: 2015/day3/ruby/mover_test.rb
require 'test/unit'
require './mover'

class MoverTest < Test::Unit::TestCase
  def test_new_location
    assert_equal([0, 1], new_location([0,0], "^"))
    assert_equal([0, -1], new_location([0,0], "v"))
    assert_equal([1, 0], new_location([0,0], ">"))
    assert_equal([-1, 0], new_location([0,0], "<"))
  end

  def test_take_directions
    assert_equal(2, take_directions(">"))
    assert_equal(4, take_directions("^>v<"))
    assert_equal(2, take_directions("^v^v^v^v^v"))
  end
end
